tsl/src/continuous_aggs/invalidation_threshold.c

```diff
@@ -259,3 +267,42 @@ continuous_agg_invalidation_threshold_lock(int32 raw_hypertable_id)
 				 errmsg("found multiple invalidation rows for hypertable %d", raw_hypertable_id)));
 	}
 }
+
+/*
+ * Compute a new invalidation threshold.
+ *
+ * The new invalidation threshold returned is the end of the given refresh
+ * window, unless it ends at "infinity" in which case the threshold is capped
+ * at the end of the last bucket materialized.
```

Comment:
Looks like this will return infinity when that's the end of the refresh window and there's no materialized data yet.  Is this correct?  Should you mention this case in the comment here?

Reply:
Good catch. Not exactly intended, not wrong either, but certainly non-optimal. I fixed this so that it returns the min time value in that case and we avoid moving the threshold unnecessarily. I added extra test cases to cover this corner-case.